vpr/src/util/vpr_utils.cpp

```diff
@@ -1933,14 +1984,15 @@ static void mark_direct_of_ports(int idirect, int direct_type, char* pb_type_nam
     auto& device_ctx = g_vpr_ctx.device();
 
     // Go through all the block types
-    for (itype = 1; itype < device_ctx.logical_block_types.size(); itype++) {
+    for (itype = 1; itype < device_ctx.physical_tile_types.size(); itype++) {
+        auto physical_tile = device_ctx.physical_tile_types[itype];
```

Comment:
`const auto & physical_tile`?

Should we delete the copy constructor on physical and logical tiles to detect unintended copies?

Reply:
I will try, even though the copy constructor is needed in the initial phase of instantiating the data structures. It will most probably cause an unintended error.

I will see if deleting the copy assignment operator is enough to do the job